File: Java/src/com/dbc/algorithm/String/IntegerToRoman.java
package com.dbc.algorithm.String;

import java.util.HashMap;
import java.util.Map;

public class IntegerToRoman {
    public String intToRoman(int num) {
        int[] values = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] symbols = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        StringBuffer result = new StringBuffer();
        for (int i = 0; i < symbols.length; i++) {
            int temp = num / values[i];
            for (int j = 0; j < temp; j++) {
                result.append(symbols[i]);
            }
            num = num % values[i];
            if (num == 0) {
                break;
            }
        }
        return result.toString();
    }
}

File: Python/string/integer_to_roman.py
def intToRoman(num: int) -> str:
    remain = {1000:"M", 900:"CM",500:"D",400:"CD",100:"C",90:"XC",50:"L",40:"XL",10:"X",9:"IX",5:"V",4:"IV",3:"III",2:"II",1:"I"}
    result = ""
    for key, value in remain.items():
        temp = num // key
        for _ in range(temp):
            result += value
        num %= key
        if num == 0:
            break
    
    return result
if __name__ == "__main__":
     print(intToRoman(3999))